MQ2AutoSize.cpp

```diff
@@ -359,269 +536,1087 @@ PLUGIN_API void OnPulse()
+			{
+				optZonewide = static_cast<int>(ResizeMode::Zonewide);
+				emulate("zonewide");
+			}
+			// General: Range
+			if (ImGui::BeginTable("OptionsResizeRangeTable", 2, ImGuiTableFlags_RowBg))
+			{
+				ImGui::TableSetupColumn("", ImGuiTableColumnFlags_WidthFixed, 20.0f);
+				ImGui::TableSetupColumn("", ImGuiTableColumnFlags_WidthFixed);
+				ImGui::TableNextColumn();
+				if (ImGui::RadioButton("", &optZonewide, static_cast<int>(ResizeMode::Range)))
+				{
+					optZonewide = static_cast<int>(ResizeMode::Range);
+					emulate("range");
+				}
+				ImGui::TableNextColumn();
+				ImGui::BeginDisabled(optZonewide == static_cast<int>(ResizeMode::Zonewide));
+				ImGui::PushItemWidth(50.0f);
+				if (ImGui::SliderInt("Range distance (recommended setting)##inputRD", &AS_Config.ResizeRange, 10, 250, "%d", ImGuiSliderFlags_NoInput|ImGuiSliderFlags_AlwaysClamp))
+				{
+					AS_Config.ResizeRange = RoundToNearestTen(AS_Config.ResizeRange);
+					if (AS_Config.OptAutoSave)
+					{
+						SaveINI("Range", true);
+					}
+				}
+				ImGui::EndDisabled();
+				ImGui::EndTable();
+			}
+			// General: Status output
+			if (ImGui::Button("Display status output"))
+			{
+				DoCommandf("/autosize status");
+			}
+			ImGui::SeparatorText("Toggles and Values");
+			if (ImGui::BeginTable("OptionsResizeSelfTable", 2, ImGuiTableFlags_RowBg))
+			{
+				ImGui::TableSetupColumn("", ImGuiTableColumnFlags_WidthFixed, 20.0f);
+				ImGui::TableSetupColumn("", ImGuiTableColumnFlags_WidthFixed);
+				// Option: Self
+				ImGui::TableNextColumn();
+				if (ImGui::Checkbox("##OptSelf", &AS_Config.OptSelf))
+				{
+					AS_Config.OptSelf = !AS_Config.OptSelf;
+					DoCommandf("/autosize self");
+				}
+				ImGui::TableNextColumn();
+				ImGui::SetNextItemWidth(50.0f);
+				ImGui::BeginDisabled(!AS_Config.OptSelf);
+				if (ImGui::SliderInt("Resize: Self##inputSS", &AS_Config.SizeSelf, 1, 30, "%d", ImGuiSliderFlags_NoInput | ImGuiSliderFlags_AlwaysClamp))
+				{
+					if (AS_Config.OptAutoSave)
+					{
+						SaveINI("SizeSelf", true);
+					}
+				}
+				ImGui::EndDisabled();
+				ImGui::TableNextRow();
+				// Option: Other PCs
+				ImGui::TableNextColumn();
+				if (ImGui::Checkbox("##OptPC", &AS_Config.OptPC))
+				{
+					AS_Config.OptPC = !AS_Config.OptPC;
+					DoCommandf("/autosize pc");
+				}
+				ImGui::TableNextColumn();
+				ImGui::SetNextItemWidth(50.0f);
+				ImGui::BeginDisabled(!AS_Config.OptPC);
+				if (ImGui::SliderInt("Resize: Other player(s) (incluldes those mounted)##inputOP", &AS_Config.SizePC, 1, 30, "%d", ImGuiSliderFlags_NoInput | ImGuiSliderFlags_AlwaysClamp))
+				{
+					if (AS_Config.OptAutoSave)
+					{
+						SaveINI("SizePC", true);
+					}
+				}
+				ImGui::EndDisabled();
+				ImGui::TableNextRow();
+				// Option: Pets
+				ImGui::TableNextColumn();
+				if (ImGui::Checkbox("##OptPet", &AS_Config.OptPet))
+				{
+					AS_Config.OptPet = !AS_Config.OptPet;
```

Comment:
Assigning the value is unnecessary. The Checkbox will do it for you (thats why you pass its address)